### DISCUSSION TITLE
TrajOpt failure signal

### DISCUSSION THREAD

**Discussion by savolla** [tesseract-robotics/tesseract_planning#226] [Q&A] [answered]
Good day! Is there any interface where I can retrieve TrajOpt failure signal? I want to know if TrajOpt failed due to a collision or something else in runtime so I can adjust planner parameters and recreate the planning server and replan my trajectory in real time. I can see the error message in the terminal like this;

```
Error:   Discrete collision detected between 'sphere_attached' and 'torch_base_link' with distance -0.000991146
...
more errors here..
...
Error:   TrajOptTaskflow Failure: 
         at line 51 in /home/savolla/tesseract_ws/src/tesseract_planning/tesseract_process_managers/src/core/utils.cpp
```

But to use it in runtime I need something like `composite_instruction.isCollided()` or `composite_instruction.isFailed()`. What class or interface should I use for this? Thanks!

**Comment by Levi-Armstrong** [comment, marked as answer]
The ProcessPlanningFuture has a member called interface which stores task infos which can be used to determine where and why things failed.

**Reply by savolla** [reply]
Thank you! In this case I needed to do only this;
```cpp
response.interface->isSuccessful();
```
